lib/apm.js

```diff
@@ -28,4 +32,104 @@ class Instrumentation extends EventEmitter {
   }
 }
 
-module.exports = Instrumentation;
+const apmLoggingEvents = {
+  command: ['commandStarted', 'commandSucceeded', 'commandFailed'],
+  sdam: [
+    'serverOpening',
+    'serverDescriptionChanged',
+    'serverHeartbeatStarted',
+    'serverHeartbeatSucceeded',
+    'serverHeartbeatFailed',
+    'serverClosed',
+    'topologyOpening',
+    'topologyClosed',
+    'topologyDescriptionChanged'
+  ]
+};
+
+class apmReadableStream extends Readable {
+  constructor(client, categories) {
+    super();
+    categories.forEach(category => {
+      attachMonitorListenerCategory(client, category, this);
+    });
+  }
+
+  _read() {
+    return;
+  }
+}
+
+function apmLoggingEventHandler(client) {
+  return function(err) {
+    if (!err) return;
+    if (!client.topology) throw err;
+    client.close(true, () => {
+      throw err;
+    });
+  };
+}
+
+function getMonitoringCategories(categories) {
+  const allCategories = Object.keys(apmLoggingEvents);
+  // enable all monitors if monitor = all
+  if (categories === 'all') return allCategories;
+  // else, ensure `monitors` is an array
+  const categoriesArray = Array.isArray(categories) ? categories : [categories];
+
+  // make sure each category is valid
+  categoriesArray.forEach(c => {
+    if (allCategories.indexOf(c) === -1) {
+      throw new TypeError(`Invalid APM monitoring category specified: ${c}`);
+    }
+  });
+
+  return categoriesArray;
+}
+
+function resolveMonitoringOutputFile(monitorOut) {
+  const defaultFileName = `${process.cwd()}/mongodb-monitoring-${new Date().toISOString()}.log`;
```

Comment:
We should probably use the `path` library to make this OS-agnostic

Reply:
I ran this in the command prompt, and found the following:

1. At least on Windows 10, the pathing issue does not occur.
2. This actually fails b/c Windows does not allow colons in file names. We need to find another format to output the date on the file in.
3. Still don't think this blocks us from releasing for now.

Reply:
it seems so easy to just choose a default filename that works across platforms, why not just do it now?